canonical-web-and-design/maas-ui
# fix(ui): fix doughnut tooltips making cards scroll

## Done

- Upgrade react-components and fix type issues.
- Remove old doughnut chart tooltips with those from react-component to make the tooltips stop making cards scroll on the KVM page.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit a kvm details page.
- Make your window smaller until there isn't enough space on the right to fit the tooltip. The tooltip should auto adjust.

## Fixes

Fixes: canonical-web-and-design/maas-ui#1584
